exchanges/bybit/bybit_websocket.go

```diff
@@ -107,467 +113,762 @@ func (by *Bybit) WsAuth(ctx context.Context) error {
+	}
+	return tickerData, nil
+}
 
-	return fmt.Errorf("unhandled stream data %s", string(respRaw))
+func (by *Bybit) wsProcessPublicTrade(assetType asset.Item, resp *WebsocketResponse) error {
+	var result WebsocketPublicTrades
+	err := json.Unmarshal(resp.Data, &result)
+	if err != nil {
+		return err
+	}
+	tradeDatas := make([]trade.Data, len(result))
+	for x := range result {
+		cp, err := currency.NewPairFromString(result[x].Symbol)
+		if err != nil {
+			return err
+		}
+		side, err := order.StringToOrderSide(result[x].Side)
+		if err != nil {
+			return err
+		}
+		tradeDatas[x] = trade.Data{
+			Timestamp:    result[x].OrderFillTimestamp.Time(),
+			CurrencyPair: cp,
+			AssetType:    assetType,
+			Exchange:     by.Name,
+			Price:        result[x].Price.Float64(),
+			Amount:       result[x].Size.Float64(),
+			Side:         side,
+			TID:          result[x].TradeID,
+		}
+	}
+	return trade.AddTradesToBuffer(by.Name, tradeDatas...)
 }
 
-func (by *Bybit) wsUpdateOrderbook(update *WsOrderbookData, p currency.Pair, assetType asset.Item) error {
-	if update == nil || (len(update.Asks) == 0 && len(update.Bids) == 0) {
-		return errors.New("no orderbook data")
+func (by *Bybit) wsProcessOrderbook(assetType asset.Item, resp *WebsocketResponse) error {
+	var result WsOrderbookDetail
+	err := json.Unmarshal(resp.Data, &result)
+	if err != nil {
+		return err
 	}
-	asks := make([]orderbook.Item, len(update.Asks))
-	for i := range update.Asks {
-		target, err := strconv.ParseFloat(update.Asks[i][0], 64)
+	cp, err := currency.NewPairFromString(result.Symbol)
+	if err != nil {
+		return err
+	}
+	asks := make([]orderbook.Item, len(result.Asks))
+	for i := range result.Asks {
+		asks[i].Price, err = strconv.ParseFloat(result.Asks[i][0], 64)
 		if err != nil {
 			return err
 		}
-		amount, err := strconv.ParseFloat(update.Asks[i][1], 64)
+		asks[i].Amount, err = strconv.ParseFloat(result.Asks[i][1], 64)
 		if err != nil {
 			return err
 		}
-		asks[i] = orderbook.Item{Price: target, Amount: amount}
 	}
-	bids := make([]orderbook.Item, len(update.Bids))
-	for i := range update.Bids {
-		target, err := strconv.ParseFloat(update.Bids[i][0], 64)
+	bids := make([]orderbook.Item, len(result.Bids))
+	for i := range result.Bids {
+		bids[i].Price, err = strconv.ParseFloat(result.Bids[i][0], 64)
 		if err != nil {
 			return err
 		}
-		amount, err := strconv.ParseFloat(update.Bids[i][1], 64)
+		bids[i].Amount, err = strconv.ParseFloat(result.Bids[i][1], 64)
 		if err != nil {
 			return err
 		}
-		bids[i] = orderbook.Item{Price: target, Amount: amount}
 	}
-	return by.Websocket.Orderbook.LoadSnapshot(&orderbook.Base{
-		Bids:            bids,
-		Asks:            asks,
-		Pair:            p,
-		LastUpdated:     update.Time.Time(),
-		Asset:           assetType,
-		Exchange:        by.Name,
-		VerifyOrderbook: by.CanVerifyOrderbook,
-	})
+	if len(asks) == 0 && len(bids) == 0 {
+		return nil
+	}
+	if resp.Type == "snapshot" {
```

Comment:
I think you should also add for [when the update == 1:](https://bybit-exchange.github.io/docs/v5/websocket/public/orderbook)
> Update ID. Is a sequence. Occasionally, you'll receive "u"=1, which is a snapshot data due to the restart of the service. So please overwrite your local orderbook

Since I can't see an example for is you receive an orderbook update type that is not snapshot with an update of `1`, I think its probably best to just add a case - just in case